services/azure-storage/src/provide_credential/client_certificate.rs

```diff
@@ -33,8 +33,8 @@ use crate::credential::Credential;
 /// Generate a unique JWT ID using timestamp and a pseudo-random component
 fn generate_jti(now: u64) -> String {
     // Use timestamp in nanoseconds + a hash of the timestamp for uniqueness
-    let nano_time = std::time::SystemTime::now()
-        .duration_since(std::time::UNIX_EPOCH)
+    let nano_time = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
```

Comment:
I think this is just `jiff::Timestamp::now().as_nanosecond()`. Although I guess that will behave differently than the code here for clocks set to a time before `1970-01-01T00:00:00Z`.

Reply:
I get your point. Although, I'd keep the current code as is and leave investigating such a refactor later.

Reply:
That said, `jiff::Timestamp::now().as_nanosecond()` returns `i128` where the current `nano_time` calculation retruns `u128`. Subtle issues often come from signed/unsigned integers .. Maybe in reality they are the same, but let's investigate it later.